org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java

```diff
@@ -0,0 +1,362 @@
+/*******************************************************************************
+ * Copyright (c) Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *******************************************************************************/
+
+package org.osgi.test.common.event;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Dictionary;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.osgi.framework.BundleEvent;
+import org.osgi.framework.Constants;
+import org.osgi.framework.FrameworkEvent;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.ServiceReference;
+import org.osgi.test.common.dictionary.Dictionaries;
+
+public interface EventPredicates {
+
+	static <E> Predicate<E> any() {
+		return element -> true;
+	}
+
+	static <E> Predicate<Optional<E>> isPresentAnd(final Predicate<E> predicate) {
+		return optional -> optional.isPresent() && predicate.test(optional.get());
+	}
+
+	static <E> Predicate<List<E>> element(int index, final Predicate<E> predicate) {
+		return elements -> elements.size() > 0 && elements.size() >= index && predicate.test(elements.get(index));
+	}
+
+	static <E> Predicate<List<E>> first(final Predicate<E> predicate) {
+		return elements -> element(0, predicate).test(elements);
+	}
+
+	static <E> Predicate<List<E>> last(final Predicate<E> predicate) {
+		return elements -> element(elements.size() - 1, predicate).test(elements);
+	}
+
+	static <E> Predicate<List<E>> all(final Predicate<E> predicate) {
+		return elements -> elements.stream()
+			.allMatch(predicate);
+	}
+
+	static <E> Predicate<List<E>> any(final Predicate<E> predicate) {
+		return elements -> elements.stream()
+			.anyMatch(predicate);
+	}
+
+	static <E> Predicate<List<E>> none(final Predicate<E> predicate) {
+		return elements -> elements.stream()
+			.noneMatch(predicate);
+	}
+
+	@SafeVarargs
+	static <E> Predicate<List<E>> ordered(final Predicate<E> predicate, final Predicate<E>... predicatesNext) {
+		return elements -> {
+
+			List<Predicate<E>> predicates = new ArrayList<>();
+			predicates.add(predicate);
+			predicates.addAll(Arrays.asList(predicatesNext));
```

Comment:
This should be `Collections.addAll(predicates, predicatesNext);`

Reply:
thx